src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesEdit.js

```diff
@@ -94,10 +97,16 @@ export default class ExternalDataSourcesEdit extends React.Component {
               dataOptions={[
                 { value: 'org.olf.kb.adapters.GOKbOAIAdapter', label: 'org.olf.kb.adapters.GOKbOAIAdapter' }
               ]}
+              disabled={isReadonly}
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.type" />}
               name={`${name}.type`}
-              required
-              validate={required}
+              required={!isReadonly}
+              validate={v => {
```

Comment:
You can one-line this fairly easily.
```suggestion
              validate={v => (value.readonly ? undefined : requiredValidator(v))}
```

Reply:
Yeah, that's a lot cleaner